### ISSUE TITLE
dbserver start does not obey the db path used + race condition with multiple processes

### ISSUE THREAD

**Issue by bebosudo** [gem/oq-engine#5472]
As the title says, `oq dbserver start host:port path` does not write to the path defined as input.
E.g. if we want to define a new dbpath:
```console
$ ls ~/oqdata/db.sqlite3*
~/oqdata/db.sqlite3*: No such file or directory
$ oq dbserver start 0.0.0.0:1234 ~/oqdata/db.sqlite3_different
...
WARNING:root:DB server started with ~/.venvs/oq_dev/bin/python3 on tcp://0.0.0.0:1234, pid 16187
$ ls ~/oqdata/db.sqlite3*
~/oqdata/db.sqlite3
```

This is because the db object that is passed at the end of the `run_server` function https://github.com/gem/oq-engine/blob/98f76f16b7374b15b717ea5e9972843a0d1313ea/openquake/server/dbserver.py#L219 is defined at the module-level at https://github.com/gem/oq-engine/blob/98f76f16b7374b15b717ea5e9972843a0d1313ea/openquake/server/dbserver.py#L36-L38 with the default config value `config.dbserver.file`.

A dirty trick to fix this is adding
```python
    db.args = (dbpath, ) + db.args[1:]
```
inside the `run_server` function at https://github.com/gem/oq-engine/blob/98f76f16b7374b15b717ea5e9972843a0d1313ea/openquake/server/dbserver.py#L183-L190

This fix now makes dbserver create the sqlite db file at the requested path:
```console
$ oq dbserver start 0.0.0.0:1234 ~/oqdata/db.sqlite3_different
$ ls ~/oqdata/db.sqlite3*
db.sqlite3_different
```
Obviously this is not the correct way to fix this. Probably changing the Db class removing the path from the args instance property and setting it only when is needed, e.g. using a `Db.set_dbpath()` method?

Reasoning:
I'm building a wrapper around oq to be able to submit multiple simulations on a HPC cluster. I've discovered that the architecture of OQ doesn't scale; in particular I get a race condition when multiple processes run in parallel: they all take the same next sequential id in the sqlite table, and when they get to the saving point, the first process is able to save, while all the others fail with the error:
```
IntegrityError: PRIMARY KEY must be unique: INSERT INTO job (id, is_running, description, user_name, calculation_mode, ds_calc_dir) VALUES (?, ?, ?, ?, ?, ?) (794, 1, 'just created', 'cscaini0', 'to be set', '~/oqdata/calc_794')
```
If you think this problem may be interesting to be fixed, I'll open a separate issue.

**Comment by micheles** [comment]
To me it looks like you are approaching the engine/cluster integration completely from the wrong side. What you need, in order to do the things properly, is the following:

1) a Python library which is able to talk to the scheduler of your cluster
2) add a new submit function in parallel.py, something like (assuming the scheduler
has a method called apply_async)
```python
@submit.add('hpc_cluster')
def hpc_cluster_submit(self, func, args, monitor):
    return self.hpc_cluster.apply_async(
        safely_call, (func, args, self.task_no, monitor))
```
3) some new configuration in openquake.cfg containing information about the address of the
scheduler and possibly other parameters.
4) a few lines of glue code in the proper places.

This would be rather easy for us to do, but we do not have access to an HPC cluster, so I think the best way to proceed further would be a collaboration project. It would probably save you time and effort.
